spring-kafka/src/main/java/org/springframework/kafka/listener/BatchLoggingErrorHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.Iterator;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+/**
+ * Simple handler that invokes a {@link LoggingErrorHandler} for each record.
+ *
+ * @author Gary Russell
+ * @since 1.1
+ *
+ */
+public class BatchLoggingErrorHandler implements BatchErrorHandler {
+
+	private final LoggingErrorHandler recordHandler = new LoggingErrorHandler();
+
+	@Override
+	public void handle(Exception thrownException, ConsumerRecords<?, ?> data) {
+		Iterator<?> iterator = data.iterator();
+		while (iterator.hasNext()) {
+			this.recordHandler.handle(thrownException, (ConsumerRecord<?, ?>) iterator.next());
```

Comment:
Why can't it just log an entire batch? And only once.
Otherwise we see a lot of StackTrace thanks to:
```
log.error("Error while processing: " + ObjectUtils.nullSafeToString(record), thrownException);
```
Isn't it?
(Sorry haven't tested)